runtime/rasdump/javadump.cpp

```diff
@@ -5165,6 +5184,65 @@ JavaCoreDumpWriter::writeCPUinfo(void)
 	return;
 }
 
+#if defined(LINUX)
+void
+JavaCoreDumpWriter::writeCgroupMetrics(void)
+{
+	OMRPORT_ACCESS_FROM_J9PORT(_PortLibrary);
+	BOOLEAN isCgroupSystemAvailable = omrsysinfo_cgroup_is_system_available();
+	if (isCgroupSystemAvailable) {
+		const OMRCgroupEntry *entryHead = omrsysinfo_get_cgroup_subsystem_list();
+		OMRCgroupEntry *cgEntry = (OMRCgroupEntry *)entryHead;
+		int32_t rc = 0;
+		if (NULL != cgEntry) {
+			_OutputStream.writeCharacters("NULL \n");
+			_OutputStream.writeCharacters("1CICGRPINFO    Cgroup Information \n");
+			_OutputStream.writeCharacters("NULL           ------------------------------------------------------------------------\n");
+			do {
+				_OutputStream.writeCharacters("2CICGRPINFO    subsystem : ");
+				_OutputStream.writeCharacters(cgEntry->subsystem);
+				_OutputStream.writeCharacters("\n");
+				_OutputStream.writeCharacters("2CICGRPINFO    cgroup name : ");
+				_OutputStream.writeCharacters(cgEntry->cgroup);
+				_OutputStream.writeCharacters("\n");
+				OMRCgroupMetricElement metricElement = {0};
+				OMRCgroupMetricIteratorState cgroupState = {0};
+				rc = omrsysinfo_cgroup_subsystem_iterator_init(cgEntry->flag, &cgroupState);
+				if (0 == rc) {
+					if (0 != cgroupState.numElements) {
+						/* print the data in columns */
+						while (0 != omrsysinfo_cgroup_subsystem_iterator_hasNext(&cgroupState)) {
+							const char *metricKey;
+							rc = omrsysinfo_cgroup_subsystem_iterator_metricKey(&cgroupState, &metricKey);
+							if (0 == rc) {
+								rc = omrsysinfo_cgroup_subsystem_iterator_next(&cgroupState, &metricElement);
+								if (rc == 0) {
+									_OutputStream.writeCharacters("2CICGRPINFO		");
+									_OutputStream.writeCharacters(metricKey);
+									_OutputStream.writeCharacters(" : ");
+									_OutputStream.writeCharacters(metricElement.value);
+									if (NULL != metricElement.units) {
+										_OutputStream.writeCharacters(" ");
+										_OutputStream.writeCharacters(metricElement.units);
+									}
+									_OutputStream.writeCharacters("\n");
+								} else {
+									_OutputStream.writeCharacters("2CICGRPINFO		");
+									_OutputStream.writeCharacters(metricKey);
+									_OutputStream.writeCharacters(" : Unavailable\n");
+								}
+							}
+						}
+						omrsysinfo_cgroup_subsystem_iterator_destroy(&cgroupState);
```

Comment:
Should this destroy call move out one level so the code would be equivalent to:
```
	rc = omrsysinfo_cgroup_subsystem_iterator_init(cgEntry->flag, &cgroupState);
 	if (0 == rc) {
 	 	.....
 	 	omrsysinfo_cgroup_subsystem_iterator_destroy(&cgroupState);
 	}
```

Reply:
api call `omrsysinfo_cgroup_subsystem_iterator_destroy` moved out one level